cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -33,11 +32,27 @@ type MasterConfiguration struct {
 type API struct {
 	AdvertiseAddresses []string `json:"advertiseAddresses"`
 	ExternalDNSNames   []string `json:"externalDNSNames"`
-	BindPort           int32    `json:"bindPort"`
+	Port               int32    `json:"port"`
 }
 
 type Discovery struct {
-	BindPort int32 `json:"bindPort"`
+	HTTPS *HTTPSDiscovery
```

Comment:
Be consistent here with all the above fields and do `json:"https"` here and below as well